### ISSUE TITLE
Ctrl C not working

### ISSUE THREAD

**Issue by jimbiladas** [wonderunit/storyboarder#1270]
Ctrl C is not working when I copy a selected board using my keyboard shortcut "ctrl C". However, if I use the dropdown menu (edit/copy) , I can do it. It will also show the confirmation "Copied Boards to Clipboard" pop up if I use it from the dropdown menu. No pop up using actual keyboard. I suppose this implies it isnt seeing my commands from the keyboard. 

As a test, I hit the "n" key to make a new board from the keyboard. And that works fine. So, some keyboard commands are working, others are not.

**Comment by audionerd** [comment]
The Keyboard Commands window has a display at the bottom which will show what keys Storyboarder thinks are pressed. Try opening that window and pressing Ctrl+C. Does it show up in the bottom display? 

Also, can you check your keymap JSON file to see if it contains the correct mapping? You can find that from Preferences.

**Comment by jimbiladas** [comment]
Hi Erik,

I don't understand what you mean by showing a keyboard commands menu. Is
this different than the one that pops up in the lower right? I looked in
preferences, and I dont see where to enable a window showing key commands.
Can you explain to me on how to do it? Is there a radio button I have to
check or something?

I'll go step by step for you:

1- I hit "Ctrl C" from the keyboard.
2- No display at the bottom right.
3- I hit Edit/Copy from the top menu, avoiding the keyboard.
4- Window display pops up in the bottom right, saying "Copied board(s) to
Clipboard"
5- I can paste using the menu at the top, going to Edit/Paste
6- Window pops up, saying "pasting a board, paste complete"

To sum up, keyboard wont let me copy, but using my mouse and using top
menu, will allow me to copy paste.

As a side note:
1- I hit "n" on my keyboard
2- new board is populated
3- no display in bottom right.
4-I hit "ctrl z"
5-board is removed.
6-no display on bottom right.

This tells me that storyboarder is seeing my keyboard when I use the "New
Board, or Undo" key command, but refuses to work with certain other key
commands. Thoughts?

Lastly, the JSON file has this:
"menu:edit:copy": "CommandOrControl+c"
"menu:edit:paste": "CommandOrControl+v"


Let me know if you need anything else.

Thanks,

James

**Comment by jimbiladas** [comment]
Hi Erik,

I don't understand what you mean by showing a keyboard commands menu. Is this different than the one that pops up in the lower right? I looked in preferences, and I dont see where to enable a window showing key commands. Can you explain to me on how to do it? Is there a radio button I have to check or something?

I'll go step by step for you:

1- I hit "Ctrl C" from the keyboard. 
2- No display at the bottom right.
3- I hit Edit/Copy from the top menu, avoiding the keyboard.
4- Window display pops up in the bottom right, saying "Copied board(s) to Clipboard"
5- I can paste using the menu at the top, going to Edit/Paste
6- Window pops up, saying "pasting a board, paste complete"

To sum up, keyboard wont let me copy, but using my mouse and using top menu, will allow me to copy paste.

As a side note:
1- I hit "n" on my keyboard
2- new board is populated
3- no display in bottom right.
4-I hit "ctrl z"
5-board is removed.
6-no display on bottom right.

This tells me that storyboarder is seeing my keyboard when I use the "New Board, or Undo" key command, but refuses to work with certain other key commands. Thoughts?

Lastly, the JSON file has this:
"menu:edit:copy": "CommandOrControl+c"
"menu:edit:paste": "CommandOrControl+v"

**Comment by audionerd** [comment]
@jimbiladas In the menu, you can go to "Help">"Key Commands" to see the Key Commands window. At the bottom of that window, it should display "Meta+c" and "menu:edit:copy" when you press Ctrl+C ("Meta" means `Control` on Windows and `Command` on Mac). If that text displays there, then it means Storyboarder understood your key press. Does that text appear when you press the key?

The JSON file contents looks correct. Thanks.

Is your keyboard mapped as United States / English?

**Comment by jimbiladas** [comment]
I went to Help>Key Commands.

I don't see a "Meta+c", screenshot attached. Also, kind of weird, why are some of the commands listed in white, and others greyed out? See attached.

My keyboard is mapped to US English, unless you are asking me to set it inside Storyboarder somewhere. If that is the case, you'll have to point out that path to me as well. I don't see a setting for that in the program.

![no_meta](https://user-images.githubusercontent.com/40301961/42351131-315263aa-8068-11e8-82e7-13da44b5d0a6.png)

**Comment by audionerd** [comment]
So you went to the Key Commands window, pressed Control+C, and didn't see anything in the bottom display? Does pressing Control or c do anything on that screen? Do other commands involving the Control key work?

**Comment by jimbiladas** [comment]
I misunderstood your request, I was assuming you wanted me to hit a button to "check" a setting to see key commands while working, as if I was going to stream for youtube etc.

Based on your reply, I hit "ctrl c" while the window is open. It does indeed recognize the keyboard.
![meta_key_command](https://user-images.githubusercontent.com/40301961/42351550-7585a5d0-806a-11e8-9c0c-3fc0de3c79c7.png)

**Comment by jimbiladas** [comment]
Hi Eric, I uninstalled the latest version of Storyboarder and went back to 1.5.1.

Copy and Paste work just dandy now. As much as I want to use the latest version, copy/paste is pretty dang important as I work. I can always make a backup and test for you later on a newer version. Just let me know.

**Comment by audionerd** [comment]
I think this is a bug in the key handler.

---

Tech notes:

`KeyboardEvent.key` returns `Control`
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values#Modifier_keys

But we’re checking for `Ctrl`
https://github.com/wonderunit/storyboarder/blob/7b49c5dcc03ab413488893dc8919edaa4f4c4bb5/src/js/utils/keytracker.js#L77

Should try checking for `Control` instead of `Ctrl` and see if that fixes this bug.

**Comment by denisvanw** [comment]
Same bug for me with the "Keys" window showing the same (and wrong) result for control C : 
menu : view : toggle-captions

**Comment by audionerd** [comment]
This will be fixed in the next release.
